packages/data/src/lib/data-view/data-view.spec.ts

```diff
@@ -0,0 +1,173 @@
+        new CartoError({
+          type: '[DataView]',
+          message: 'Source was not provided while creating dataview'
+        })
+      );
+    });
+
+    it('should throw an exception when column is not provided', () => {
+      expect(
+        () => new DataView(new Layer('fake_source'), undefined as never)
+      ).toThrow(
+        new CartoError({
+          type: '[DataView]',
+          message: 'Column name was not provided while creating dataview'
+        })
+      );
+    });
+  });
+
+  describe('Events', () => {
+    it('dataUpdate', () => {
+      const layer = new Layer('fake_source');
+      const dataView = new DataView(layer, 'popEst');
+
+      const dataUpdateSpy = jest.fn();
+      dataView.on('dataUpdate', dataUpdateSpy);
+
+      layer.emit('viewportLoad');
+
+      expect(dataUpdateSpy).toHaveBeenCalled();
+    });
+  });
+
+  describe('aggregate', () => {
+    it('should return an empty object if operations are not provided', () => {
+      const dataView = new DataView(new Layer('fake_source'), 'fake_column');
+      expect(dataView.aggregate()).toMatchObject({});
+    });
+
+    it('should return an object with data aggregations', async () => {
+      const layer = new Layer('fake_source');
+      spyOn(layer, 'getViewportFeatures').and.returnValue(
+        Promise.resolve(sourceData)
+      );
+
+      const dataView = new DataView(layer, 'popEst');
+
+      const aggregations = [
+        AggregationTypes.AVG,
+        AggregationTypes.COUNT,
+        AggregationTypes.MAX,
+        AggregationTypes.MIN,
+        `percentile_90` as AggregationTypes,
+        AggregationTypes.SUM
+      ];
+
+      const aggregationsResult = {
+        avg: 30,
+        count: 5,
+        max: 50,
+        min: 10,
+        sum: 150,
+        percentile_90: 50 // eslint-disable-line
+      };
+
+      expect(await dataView.aggregate(...aggregations)).toMatchObject(
+        aggregationsResult
+      );
+    });
+
+    it('should throw an error when there are no numeric values', async () => {
+      const layer = new Layer('fake_source');
+      spyOn(layer, 'getViewportFeatures').and.returnValue(
+        Promise.resolve([{ notNumericColumn: 'notprovided' }])
+      );
+
+      const dataView = new DataView(layer, 'popEst');
+
+      await expect(() =>
+        dataView.aggregate(AggregationTypes.AVG)
+      ).rejects.toThrow(
+        new CartoError({
+          type: '[DataView]',
+          message: "Provided column doesn't exist or has no numeric values"
+        })
+      );
+    });
+  });
+
+  describe('groupBy', () => {
+    it('should return average of grouped values by column', async () => {
+      const layer = new Layer('fake_source');
+      spyOn(layer, 'getViewportFeatures').and.returnValue(
+        Promise.resolve(sourceDataToGroup)
+      );
+
+      const dataView = new DataView(layer, 'popEst');
+
+      expect(await dataView.groupBy('country')).toEqual([
```

Comment:
I find a bit arbitrary to use AVG by default, without been explicit. In some variables SUM could also be a perfectly good default... In any case, I think the operation param should be mandatory, to avoid misunderstandings. Don't you think?